src/libCom/test/epicsTimerTest.cpp

```diff
@@ -27,6 +35,150 @@
 #define verify(exp) ((exp) ? (void)0 : \
     epicsAssert(__FILE__, __LINE__, #exp, epicsAssertAuthor))
 
+#if __cplusplus >= 201103L
+const double timeTolerance { 1e-3 + epicsThreadSleepQuantum() };
+
+class handler : public epicsTimerNotify
+{
+public:
+  handler(epicsTimerQueue& queue, std::function<void()> expireFctIn = [] {}) : timer(queue.createTimer()), expireFct(expireFctIn) {}
+  ~handler() { timer.destroy(); }
+  void start(double delay) {
+    startTime = epicsTime::getCurrent();
+    timer.start(*this, delay);
+  }
+  void cancel() { timer.cancel(); }
+  expireStatus expire(const epicsTime& currentTime) {
+    expireTime = epicsTime::getCurrent();
+    expireCount++;
+    expireFct();
+    return expireStatus(noRestart);
+  }
+  int getExpireCount() const { return expireCount; }
+  double getDelay() const { return expireTime - startTime; }
+private:
+  epicsTimer &timer;
+  int expireCount { 0 };
+  epicsTime startTime;
+  epicsTime expireTime;
+  std::function<void()> expireFct;
+};
+
+void verifyExpirationTime(double delta_t, double tolerance) {
+  if (!testOk(std::abs(delta_t) < tolerance, "timer expired at the expected "
```

Comment:
This tight unsigned test seems problematic.  Based on discussions elsewhere, I think we are in agreement that this delay should never by negative?  At the same time, on shared CI builder services, it could easily be longer.  Would a better pass condition be: `delta >= 0.0 && (testImpreciseTiming() || delta < tolerance)`?

Reply:
Also, to help with troubleshooting CI build failures, tests like this one should print the values being tested.

Reply:
... which I just noticed is being done on the next line (which was cropped out).

Reply:
This PR is for 3.15, which doesn't have `testImpreciseTiming()`.

Reply:
Shouldn't this addition be going into 7.0?

Reply:
Done. The 8 failing tests on Linux are due to #106 and will go away when #107 gets merged. For now I didn't include the `testImpreciseTiming()` part to maintain compatibility with 3.15. I'm happy to add that once this change has been merged up to 7.0.